android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java

```diff
@@ -190,6 +198,21 @@ private static String readCustomPwsEndpointApiKey(Context context) {
     return sharedPref.getString(context.getString(R.string.custom_pws_api_key_key), "");
   }
 
+  private static boolean restoreMainPrefSettings(Context context) {
```

Comment:
I don't think this should be its own method...just copy this code down to the method where you use it.